test/test-network/systemd-networkd-tests.py

```diff
@@ -79,8 +87,39 @@ def setUpModule():
 
     subprocess.check_call('systemctl stop systemd-networkd.socket', shell=True)
 
+    drop_in = [
+        '[Service]',
+        'Restart=no',
+        'ExecStart=',
+    ]
+    if use_valgrind:
+        drop_in += [
+            'ExecStart=!!valgrind --track-origins=yes --leak-check=full --show-leak-kinds=all ' + networkd_bin,
+            'PrivateTmp=yes',
+            'MemoryDenyWriteExecute=no',
```

Comment:
This option is necessary for ASan and UBSan runs as well. Could you, please, add something like

```python
if asan_options or ubsan_options:
    drop_in += ['MemoryDenyWriteExecute=no']
```

cc @evverx

Reply:
Well, `MemoryDenyWriteExecute` helped to find out that the test doesn't fail when `networkd` exits with a non-zero exit code (which kind of makes it impossible to run it under ASan+UBsan and Valgrind automatically because looking at the logs or grepping reports manually isn't suitable for CI). I'd leave it there for the time being. At least until the test is fixed.

Reply:
On the bright side, LSan works fine on the client side and has just discovered a memory leak:
```sh
test_lldp (__main__.NetworkdLLDPTests) ... 
=================================================================
==8883==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 12 byte(s) in 1 object(s) allocated from:
    #0 0x7f1a77b5c389 in __interceptor_malloc /build/gcc/src/gcc/libsanitizer/asan/asan_malloc_linux.cc:86
    #1 0x558315025fcb in malloc_multiply ../src/basic/alloc-util.h:96
    #2 0x558315034878 in lldp_capabilities_to_string ../src/network/networkctl.c:1167
    #3 0x5583150363fc in link_lldp_status ../src/network/networkctl.c:1312
    #4 0x7f1a76c06803 in dispatch_verb ../src/shared/verbs.c:119
    #5 0x558315037b97 in networkctl_main ../src/network/networkctl.c:1492
    #6 0x558315037c73 in run ../src/network/networkctl.c:1516
    #7 0x558315037ca6 in main ../src/network/networkctl.c:1519
    #8 0x7f1a7566ace2 in __libc_start_main (/lib64/libc.so.6+0x23ce2)

SUMMARY: AddressSanitizer: 12 byte(s) leaked in 1 allocation(s).

ERROR
```

Reply:
Fair enough, I wasn't sure if it doesn't mask some other issues :-) I'll open a separate issue for the memory leak, if you already haven't done so.

Reply:
> Fair enough, I wasn't sure if it doesn't mask some other issues :-)

@mrc0mmand it's possible to try to fix everything at once in parallel, if you can keep track of all the issues that pop up :-) Unfortunately, my bandwidth is limited and I kind of got lost when https://github.com/systemd/systemd/pull/12696 was merged and especially when we started to discuss whether the test should be installed :-)